database/postgres/event.go

```diff
@@ -388,8 +389,8 @@ func (e *eventRepo) FindEventsByIdempotencyKey(ctx context.Context, projectID st
 		return nil, err
 	}
 
-	query = e.db.Rebind(query)
-	rows, err := e.db.QueryxContext(ctx, query, args...)
+	query = e.db.GetDB().Rebind(query)
+	rows, err := e.db.GetDB().QueryxContext(ctx, query, args...)
```

Comment:
```suggestion
	query = e.db.GetReadDB().Rebind(query)
	rows, err := e.db.GetReadDB().QueryxContext(ctx, query, args...)
```

Reply:
This seems okay, but it is used by `CreateEvent`, as well as broadcast, dynamic, and fanout. And since the method is `FindEventsByIdempotencyKey`, we can assume the query will be fast if we retain the primary DB.